src/protocol.rs

```diff
@@ -450,6 +470,66 @@ mod tests {
     use std::collections::HashMap;
     use tracing::debug;
 
+    #[test]
+    fn participant_config_has_more_than_one_participant() {
```

Comment:
suggestion: this test is doing the slightly weird thing of testing a test-only function. I'd try to put that in the name, maybe like
```rust
fn random_quorum_must_have_at_least_two_participants()
```